Source/WebCore/dom/Document.cpp

```diff
@@ -1541,6 +1541,11 @@ void Document::setDocumentElementLanguage(const AtomString& language)
         return;
     m_documentElementLanguage = language;
 
+    for (auto& element : copyToVector(m_elementsWithLangAttrMatchingDocument)) {
+        if (element)
+            element->setEffectiveLangInSubtree();
```

Comment:
Don't we need to upset the flag on every element here?

Also, this set is now stale, right? There will be another set of elements that matches the new document element's lang value.

Reply:
The flag is updated and the element removed from the WeakHashSet by `setEffectiveLangInSubtree`.

But you are right that the set will be stale in that there may now be elements in the document that have a `lang=""` that matches the new document element lang, but which are not in the WeakHashSet or have the right bits set. This is OK, since on those elements we still have the effective lang stored on the ElementRareData. We could be clearer about this by calling the flag `EffectiveLangKnownToMatchDocumentElement` or something.

Reply:
>The flag is updated and the element removed from the WeakHashSet by setEffectiveLangInSubtree.

Though it would be slightly more efficient to clear the WeakHashSet here, as we'll avoid churning the table as we process each element.